tools/dotcms-cli/cli/src/main/java/com/dotcms/cli/command/EntryCommand.java

```diff
@@ -61,30 +66,106 @@ class CustomConfiguration {
+        // Injecting custom push mixins to the global push command
+        configurationUtil.injectPushMixins(cmdLine);
+        // Customizing the CommandLine object
+        configurationUtil.customize(cmdLine);
+
+        return cmdLine;
+    }
+
+}
+
+/**
+ * A utility class for customizing configurations.
+ */
+class CustomConfigurationUtil {
+
+    private static final CustomConfigurationUtil INSTANCE = new CustomConfigurationUtil();
+
+    private CustomConfigurationUtil() {
+        if (INSTANCE != null) {
+            throw new IllegalStateException("Singleton already constructed");
         }
+    }
+
+    /**
+     * Returns the singleton instance of CustomConfigurationUtil.
+     *
+     * @return the singleton instance of CustomConfigurationUtil
+     */
+    public static CustomConfigurationUtil getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Customizes a CommandLine object.
+     *
+     * @param cmdLine the CommandLine object to customize
+     * @return the customized CommandLine object
+     */
+    public void customize(CommandLine cmdLine) {
+
+        cmdLine.setCaseInsensitiveEnumValuesAllowed(true)
+                .setExecutionStrategy(
+                        new LoggingExecutionStrategy(new CommandLine.RunLast()))
+                .setExecutionExceptionHandler((ex, commandLine, parseResult) -> {
+                    final Object object = commandLine.getCommand();
+                    if (object instanceof DotCommand) {
+                        final DotCommand command = (DotCommand) object;
+                        final OutputOptionMixin output = command.getOutput();
+                        return output.handleCommandException(ex,
+                                String.format("Error in command [%s] with message: %n ",
+                                        command.getName()));
+                    } else {
+                        commandLine.getErr().println(ex.getMessage());
+                    }
+                    return ExitCode.SOFTWARE;
+                }).setExitCodeExceptionMapper(t -> {
+                    // customize exit code
+                    // We usually throw an IllegalArgumentException to denote that an invalid param has been passed
+                    if (t instanceof ParameterException || t instanceof IllegalArgumentException) {
+                        return ExitCode.USAGE;
+                    }
+                    return ExitCode.SOFTWARE;
+                });
+    }
+
+    /**
+     * Injects custom push mixins into the global push command.
+     *
+     * @param cmdLine the main entry command
+     */
+    public void injectPushMixins(CommandLine cmdLine) {
+
+        // Looking for the global push command
+        CommandSpec pushCommandSpec = cmdLine.getSubcommands().get(PushCommand.NAME)
+                .getCommandSpec();
+
+        // Get all instances that implement DotPush
+        Instance<DotPush> dotPushCommands = CDI.current().select(DotPush.class);
```

Comment:
I would encapsulate this into a class instead of hitting the API, so change in the future become an issue